python/pmmlserver/pmmlserver/model.py

```diff
@@ -57,10 +59,26 @@ def load(self) -> bool:
         self.ready = True
         return self.ready
 
-    def predict(self, payload: Dict, headers: Dict[str, str] = None) -> Dict:
-        instances = payload["instances"]
+    def predict(self, payload: Union[Dict, InferRequest], headers: Dict[str, str] = None) -> Union[Dict, InferResponse]:
         try:
-            result = [self.evaluator.evaluate(dict(zip(self.input_fields, instance))) for instance in instances]
-            return {"predictions": result}
+            if isinstance(payload, Dict):
+                instances = payload["instances"]
+                result = [self.evaluator.evaluate(
+                    dict(zip(self.input_fields, instance))) for instance in instances]
+                return {"predictions": result}
+            elif isinstance(payload, InferRequest):
+                result = [self.evaluator.evaluate(
+                    dict(zip(self.input_fields, instance))) for instance in payload.inputs[0].data]
+                infer_output = InferOutput(
+                    name="output-0",
+                    shape=[len(result)],
+                    datatype="MIXED",
```

Comment:
datatype should be converted from result

Reply:
@yuzisun How can we get the type from the result? I hope the result will be a list.

Reply:
here should be the data type of the values in the list, does `evaluate` return the numpy array?

Reply:
added a commit to convert the response to pandas and then multiple infer outputs